### PR TITLE
CMMD dataset and baseline example

### PR DESCRIPTION
**Pull request by itaijj**
added MG_CMMD dataset example and relevant new modules and tests

### CODE REVIEW

File: fuse_examples/classification/MG_CMMD/runner.py [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+
+from fuse.metrics.classification.metric_auc import FuseMetricAUC
+from fuse.metrics.classification.metric_accuracy import FuseMetricAccuracy
+
+from fuse.managers.callbacks.callback_tensorboard import FuseTensorboardCallback
+from fuse.managers.callbacks.callback_metric_statistics import FuseMetricStatisticsCallback
+from fuse.managers.callbacks.callback_time_statistics import FuseTimeStatisticsCallback
+from fuse.managers.manager_default import FuseManagerDefault
+
+#from fuse_examples.classification.MG_CMMD.dataset import CMMD_2021_dataset
```
Comment (mosheraboh) [inline fuse_examples/classification/MG_CMMD/runner.py:48]: remove

File: fuse_examples/classification/MG_CMMD/runner.py [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+Created on June 30, 2021
+
+"""
+
+import os
+
+from fuse.utils.utils_debug import FuseUtilsDebug
+from fuse.utils.utils_gpu import FuseUtilsGPU
+
+os.environ['skip_broker'] = '1'
```
Comment (mosheraboh) [inline fuse_examples/classification/MG_CMMD/runner.py:23]: remove, not required any more

File: fuse_examples/classification/MG_CMMD/dataset_folds.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
+import sys
```
Comment (mosheraboh) [inline fuse_examples/classification/MG_CMMD/dataset_folds.py:1]: lets rename the file to dataset.py

File: fuse_examples/classification/MG_CMMD/data_source_folds.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
+"""
```
Comment (mosheraboh) [inline fuse_examples/classification/MG_CMMD/data_source_folds.py:1]: this one is generic, right?
lets move the file to fuse/data/data_source/

**Review by mosheraboh — COMMENTED** [review]
Great!
Adress the inline comments and then you can merge to master.
There was another runner that should go to FuseDev, right?

File: fuse_examples/classification/MG_CMMD/dataset_folds.py [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+            scans.append({'ID1': patient, 'LeftRight': dcm[0x00200062].value, 'file': file,
+                          'view': dcm[0x00540220].value.pop(0)[0x00080104].value})
+    dicom_tags = pd.DataFrame(scans)
+    merged_clinical_data = pd.merge(clinical_data, dicom_tags, how='outer', on=['ID1', 'LeftRight'])
+    merged_clinical_data = merged_clinical_data[merged_clinical_data[target].notna()]
+    merged_clinical_data.to_csv(combined_file_path)
+    return combined_file_path
+
+
+if __name__ == "__main__":
```
Comment (mosheraboh) [inline fuse_examples/classification/MG_CMMD/dataset_folds.py:172]: Lets also remove all the unused mains. It's a bit confusing (we should remove it from old files as well)

File: fuse_examples/classification/MG_CMMD/runner.py (hunk 1 above)
Comment (itaijj) [reply fuse_examples/classification/MG_CMMD/runner.py:48]: done

File: fuse_examples/classification/MG_CMMD/runner.py (hunk 2 above)
Comment (itaijj) [reply fuse_examples/classification/MG_CMMD/runner.py:23]: done

File: fuse_examples/classification/MG_CMMD/dataset_folds.py (hunk 3 above)
Comment (itaijj) [reply fuse_examples/classification/MG_CMMD/dataset_folds.py:1]: done

File: fuse_examples/classification/MG_CMMD/data_source_folds.py (hunk 4 above)
Comment (itaijj) [reply fuse_examples/classification/MG_CMMD/data_source_folds.py:1]: done , added changes to the file on summary and toolbox

File: fuse_examples/classification/MG_CMMD/dataset_folds.py (hunk 5 above)
Comment (itaijj) [reply fuse_examples/classification/MG_CMMD/dataset_folds.py:172]: done

File: fuse_examples/classification/MG_CMMD/dataset.py [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+    test_dataset = FuseDatasetDefault(cache_dest=cache_dir,
+                                            data_source=test_dataset_source,
+                                            input_processors=input_processors,
+                                            gt_processors=gt_processors,
+                                            post_processing_func=post_cache_processing_func,
+                                            augmentor=None,
+                                            visualizer=visualiser)
+
+    lgr.info(f'- Load and cache data:')
+    validation_dataset.create(
```
Comment (mosheraboh) [inline fuse_examples/classification/MG_CMMD/dataset.py:144]: This should to be right after you create the validation dataset.
And here you missed test_dataset.create(..)

File: fuse_examples/classification/MG_CMMD/dataset.py [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+            scans.append({'ID1': patient, 'LeftRight': dcm[0x00200062].value, 'file': file,
+                          'view': dcm[0x00540220].value.pop(0)[0x00080104].value})
+    dicom_tags = pd.DataFrame(scans)
+    merged_clinical_data = pd.merge(clinical_data, dicom_tags, how='outer', on=['ID1', 'LeftRight'])
+    merged_clinical_data = merged_clinical_data[merged_clinical_data[target].notna()]
+    merged_clinical_data.to_csv(combined_file_path)
+    return combined_file_path
+
+
+if __name__ == "__main__":
```
Comment (mosheraboh) [inline fuse_examples/classification/MG_CMMD/dataset.py:179]: remove the main here also

File: fuse_examples/classification/MG_CMMD/input_processor.py [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+    cells_binary[cells>dark_area_threshold] = True #TODO: this is a hardcoded threshold, see if there's a better alternative
+    blobs_labels = measure.label(cells_binary, background=0)
+    regions = measure.regionprops(blobs_labels)
+
+    regions_areas = [r.area for r in regions]
+    if len(regions)<1:
+        # expected to have at least background and one object. TODO: consider reporting error and returning AABB of the full image.
+        print('Warning: could not crop properly! fallbacking to full image')
+        return 0,0,img.shape[0]-1,img.shape[1]-1
+        #raise Exception('expected 2 regions')
```
Comment (mosheraboh) [inline fuse_examples/classification/MG_CMMD/input_processor.py:120]: remove

**Review by mosheraboh — CHANGES_REQUESTED** [review]
Thanks, two more comments inline

File: fuse_examples/classification/MG_CMMD/dataset.py (hunk 6 above)
Comment (itaijj) [reply fuse_examples/classification/MG_CMMD/dataset.py:144]: ok, fixed

File: fuse_examples/classification/MG_CMMD/dataset.py (hunk 7 above)
Comment (itaijj) [reply fuse_examples/classification/MG_CMMD/dataset.py:179]: done

File: fuse_examples/classification/MG_CMMD/input_processor.py (hunk 8 above)
Comment (itaijj) [reply fuse_examples/classification/MG_CMMD/input_processor.py:120]: done

**Review by mosheraboh — APPROVED** [review]

**Pull request merged**